### PR TITLE
Add ESLint + Prettier

### PR DESCRIPTION
**Pull request by hbergren**
## High Level Overview of Change

Add the standardized JS/TS linting configuration to `xpring-common-js`.

### Notes for Reviewers

My commits are way less organized than this PR for `xpring-common-js`, but the idea is the same.

### Context of Change

The goal here is to get all of our JS/TS projects on the same configuration so we have some consistency across projects. This linting configuration is live on `ripple-keypairs` and will be migrated to the other `ripple-lib` style libraries moving forward.

Next steps after this PR is perhaps the wallet (with some tweaks for Jest + React).

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)

## Before / After

Now we have a more robust linting configuration, and Prettier for auto-formatting. Some linting errors were also fixed.

## Test Plan

Run the tests, they all still pass.

### CODE REVIEW

**Comment by hbergren** [comment]
@keefertaylor , ESLint is pretty upset about the emptiness of `default-xpring-client`. I'm not sure what to do about that.

We could disable the violating ESLint rules like you do at the top under:
```
// TODO(keefertaylor): Re-enable this rule when this class is fully implemented.
/* eslint-disable @typescript-eslint/require-await */
```

**Comment by keefertaylor** [comment]
That would be fine with me. I have a stack of PRs that I'd like to put out to start implementing this stuff but it will probably take at least a week. We can either hold this PR or disable the rules. Up to you. 

If you want to disable rules, could you file an Asana issue?

File: test/fakes/fake-xpring-client.ts [hunk 1]
```diff
@@ -1,38 +1,44 @@
...
 
-    public async getLastValidatedLedgerSequence(): Promise<number> {
-        return Promise.resolve(this.getLastValidatedLedgerSequenceValue)
-    }
-    
-    public async getRawTransactionStatus(transactionHash: string): Promise<RawTransactionStatus> {
-       return Promise.resolve(this.getRawTransactionStatusValue);
-    }
+  public async send(
+    _amount: any,
```
Comment (Stormtv) [inline test/fakes/fake-xpring-client.ts]:
```suggestion
    _amount: BigInt | number | string,
```

File: src/default-xpring-client.ts [hunk 2]
```diff
@@ -1,9 +1,7 @@
...
-import { Wallet } from "xpring-common-js";
-import { TransactionStatus as RawTransactionStatus } from "../generated/legacy/transaction_status_pb";
-
-/* global BigInt */
+import { Wallet } from 'xpring-common-js'
+import { XpringClientDecorator } from './xpring-client-decorator'
+import TransactionStatus from './transaction-status'
+import { TransactionStatus as RawTransactionStatus } from '../generated/legacy/transaction_status_pb'
 
 // TODO(keefertaylor): Re-enable this rule when this class is fully implemented.
```
Comment (Stormtv) [inline src/default-xpring-client.ts]: I think we should just disable these under the TODO to re-enable these rules when completed
```suggestion
 // TODO(keefertaylor): Re-enable these rules when this class is fully implemented.
/* eslint-disable @typescript-eslint/require-await */
/* eslint-disable @typescript-eslint/no-empty-function */
/* eslint-disable @typescript-eslint/no-useless-constructor */
/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable class-methods-use-this */
/* eslint-disable max-classes-per-file */
```

File: src/default-xpring-client.ts [hunk 3]
```diff
@@ -1,9 +1,7 @@
-import { XpringClientDecorator } from "./xpring-client-decorator";
-import TransactionStatus from "./transaction-status";
-import { Wallet } from "xpring-common-js";
-import { TransactionStatus as RawTransactionStatus } from "../generated/legacy/transaction_status_pb";
-
-/* global BigInt */
+import { Wallet } from 'xpring-common-js'
```
Comment (Stormtv) [inline src/default-xpring-client.ts]:
```suggestion
/* eslint-disable max-classes-per-file */
import { Wallet } from 'xpring-common-js'
```

**Comment by hbergren** [comment]
I pulled in all the changes Tyler suggested, with one exception.

ESLint doesn't like defining multiple classes in the same file like you do for `default-xpring-client` with the helper `class XpringClientErrorMessages`.

i think in general that the 1 class per file thing is a good rule, but for this tiny helper class, I don't really care. I want to treat inline `/* eslint-disable */` rules as things we rip out, but since the 2 class per file for `xpring-default-client` is a reasonable exception, I added that directly to the `.eslintrc.js` config file.

I made an Asana task for re-enabling all the ESLint rules for this file in the backlog:
https://app.asana.com/0/1143722888069470/1158300650916617/f

**Pull request merged**